pkg/redirectpolicy/manager.go

```diff
@@ -0,0 +1,531 @@
+			// pending request.
+			continue
+		}
+		if !config.policyConfigSelectsPod(podMetadata) {
+			continue
+		}
+		retEps = append(retEps, podMetadata)
+	}
+
+	return retEps
+}
+
+// Returns backend entries for pod(s) for the given redirect policy config.
+func (rpm *RedirectPolicyManager) getPodBackendsForPolicy(config *LRPConfig, podsInfo ...*epPodInfo) []backend {
+	var backends []backend
+
+	for _, pod := range podsInfo {
+		podIP := pod.ip
+		bProto := lb.L4Type(config.backendPortProto.Protocol)
+		// Port and Protocols are already sanitized.
+		// TODO add a unit test case
+		bPort := config.backendPortProto.Port
+		if iana.IsSvcName(bPort) {
+			// Pod backends are matched based on the config port name
+			if podBackend, found := rpm.getPodBackendWithNamedPort(pod, bPort); found {
+				backends = append(backends, podBackend)
+			}
+		} else if isSubset, _ := set.SliceSubsetOf([]lb.L4Type{bProto}, lb.AllProtocols); isSubset {
+			// Pod backends are matched based on the config L4 port and protocol
+			if podBackend, found := rpm.getPodBackendWithProtocol(podIP, bProto, bPort); found {
+				backends = append(backends, podBackend)
+			}
+		} else if strings.EqualFold(bProto, string(api.ProtoAny)) {
+			// Pod backends are matched based on all the supported L4 port and protocol
+			for i := range lb.AllProtocols {
+				if lb.AllProtocols[i] != config.frontend.Protocol {
+					// Remove this once we support multiple ports for address/service matchers.
+					continue
+				}
+				if podBackend, found := rpm.getPodBackendWithProtocol(podIP, lb.AllProtocols[i], bPort); found {
+					backends = append(backends, podBackend)
+				}
+			}
+		}
+		if len(backends) > 0 {
+			// Add this pod as a backend in the given policy config.
+			rpm.podPolicy[podIP] = append(rpm.podPolicy[podIP], podPolicyInfo{
+				policyID: config.getPolicyID(),
+				backends: backends,
+			})
+		}
+	}
+
+	return backends
+}
+
+// isValidConfig validates the given policy config.
+func (rpm *RedirectPolicyManager) isValidConfig(config LRPConfig) (bool, error) {
+	if config.frontend != nil {
+		name, ok := rpm.frontends[config.frontend.Hash()]
+		if ok && config.name != name {
+			return false, fmt.Errorf("CiliumLocalRedirectPolicy for"+
+				"IPPort %v already exists", config.frontend)
+		}
+	}
+
+	if config.serviceID != nil {
+		p, ok := rpm.services[config.serviceID.String()]
+		if ok && config.namespace == rpm.policyConfigs[p].namespace {
+			return false, fmt.Errorf("CiliumLocalRedirectPolicy for"+
+				" service %v already exists", config.serviceID)
+		}
+	}
+
+	return true, nil
+}
+
+// Event handlers
+
+// AddRedirectPolicy parses the given local redirect policy config, and updates
+// internal state with the config fields.
+func (rpm *RedirectPolicyManager) AddRedirectPolicy(config LRPConfig, svcCache *k8s.ServiceCache) (bool, error) {
+	rpm.Lock()
+	defer rpm.Unlock()
+
+	_, ok := rpm.policyConfigs[config.getPolicyID()]
+	if ok {
+		// TODO Existing policy update
```

Comment:
Will you address this TODO in this PR? If not it might be good to add a warning here stating clearly that the update for an existing policy is being ignored.